File: Insertion Sort/insertionsort.cpp
#include <bits/stdc++.h>

using namespace std;

void insertionsort(vector<int>& v){
    int chave, i;
    
    for (int j = 0; j < v.size(); j++)
    {
        chave = v[j];
        i = j -1;
        while (i >= 0 and v[i] > chave)
        {
            v[i+1] = v[i];
            i--;
        }
        v[i+1] = chave;
    }
    
}

int main()
{
    vector<int> v = {4,2,3,1};
    insertionsort(v);
    for (int i = 0; i < v.size(); i++)
    {
        cout << v[i] << " ";
    }
    cout << endl;
    
}

File: Insertion Sort/insertionsort.py
'''
    @author: Guilherme Vieira Silva Gonçalves
    @course: Ciência da Computação, 5th semester
'''

def insertionsort(v):
    for j in range(0, len(v)):
        chave = v[j]
        i = j - 1
        while(i >= 0 and v[i] > chave):
            v[i + 1] = v[i]
            i = i - 1
        v[i + 1] = chave
    return v
        
def main():
    v = [3,2,1,6]
    print(insertionsort(v))

if __name__ == "__main__":
    main()
            
